cpp/tests/io/json_test.cpp

```diff
@@ -1704,8 +1734,10 @@ TYPED_TEST(JsonFixedPointReaderTest, EmptyValues)
 TEST_F(JsonReaderTest, UnsupportedMultipleFileInputs)
 {
   std::string const data = "{\"col\":0}";
-  auto const buffer      = cudf::io::host_buffer{data.data(), data.size()};
-  auto const src         = cudf::io::source_info{{buffer, buffer}};
+  auto const buffer =
+    cudf::host_span<std::byte const>(reinterpret_cast<std::byte const*>(data.data()), data.size());
+  auto const src =
+    cudf::io::source_info{std::vector<cudf::host_span<std::byte const>>{buffer, buffer}};
```

Comment:
Very disappointed that we explicitly need to specify the type here. What is the error without the change in this line?

Reply:
I agree entirely. With the new API's I just added we may be able to get away with something better here. I will test.

```
error: no matching function for call to 'cudf::io::source_info::source_info(<brace-enclosed initializer list>)'
 1740 |     cudf::io::source_info{{buffer, buffer}};
```

I was able to compile and tests passed with
```
cudf::io::source_info{std::vector{buffer, buffer}}; 
```